File: 3683-find-the-lexicographically-largest-string-from-the-box-i/find-the-lexicographically-largest-string-from-the-box-i.java
class Solution {
    public String answerString(String word, int numFriends) {
        if(numFriends == 1) return word;

        int n = word.length(), end;
        int split = n-numFriends;

        String curr = "", m = "";

        for(int i=0; i<n; i++){
            end = Math.min(n, i+split +1);
            curr = word.substring(i, end);
            if(curr.compareTo(m) > 0)
            m = curr;
        }
        return m;
    }
}

File: 3683-find-the-lexicographically-largest-string-from-the-box-i/find-the-lexicographically-largest-string-from-the-box-i.js
/**
 * @param {string} word
 * @param {number} numFriends
 * @return {string}
 */
var answerString = function (word, numFriends) {
    if (numFriends == 1)
        return word;
    let n = word.length;
    let split_length = n - numFriends, m = '', cur = '';
    for (let i = 0; i < n; i++) {
        cur = word.slice(i, i + split_length + 1);
        if (cur > m) {
            m = cur;
        }
    }
    return m;
};